Panic in tests: Log in goroutine after TestStreamingBouncer has completed

There is a test lifecycle race condition in TestStreamingBouncer and potentially other bouncer tests.

**Steps to reproduce:**
```
go test -v -race ./internal/bouncer -run TestStreamingBouncer -count=10
```

Received:
```
=== RUN   TestStreamingBouncer
    logger.go:146: 2026-02-19T20:09:04.146+0100	INFO	started	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.146+0100	DEBUG	starting metrics provider	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.147+0100	DEBUG	starting decision processing	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.147+0100	DEBUG	starting streaming bouncer	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.147+0100	DEBUG	processing 5 new decisions	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.147+0100	DEBUG	adding "127.0.0.1" (scope: Ip) for "120s"	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.148+0100	DEBUG	adding "127.0.0.2" (scope: Ip) for "120s"	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.148+0100	DEBUG	adding "10.0.0.1/24" (scope: Range) for "120s"	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.148+0100	DEBUG	adding "128.0.0.1/32" (scope: Ip) for "120s"	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.148+0100	ERROR	unable to insert decision for "129.0.0.1/24": 129.0.0.1/24 seems to be a range instead of an IP	{"instance_id": "3d28daf4"}
    logger.go:146: 2026-02-19T20:09:04.148+0100	DEBUG	finished processing 5 new decisions	{"instance_id": "3d28daf4"}
--- PASS: TestStreamingBouncer (1.03s)
panic: Log in goroutine after TestStreamingBouncer has completed: 2026-02-19T20:09:05.149+0100	INFO	metrics provider stopped	{"instance_id": "3d28daf4"}
	

goroutine 28 [running]:
testing.(*common).log(0xc0002438c8, {0xc000064240, 0x56})
	/opt/homebrew/Cellar/go/1.26.0/libexec/src/testing/testing.go:1039 +0x178
testing.(*common).Logf(0xc0002438c8, {0x103f9ca6d, 0x2}, {0xc00050b170, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.26.0/libexec/src/testing/testing.go:1200 +0x80
go.uber.org/zap/zaptest.TestingWriter.Write({{0x1058d1cd0?, 0xc0002438c8?}, 0x60?}, {0xc000692c00, 0x57, 0x400})
	/Users/richardkapolnai/go/pkg/mod/go.uber.org/zap@v1.27.0/zaptest/logger.go:146 +0xf0
go.uber.org/zap/zapcore.(*ioCore).Write(0xc0002f58c0, {0x0, {0xc25df5d448e1c5f0, 0x42b43b77, 0x105a409c0}, {0x0, 0x0}, {0x103fc62f2, 0x18}, {0x0, ...}, ...}, ...)
	/Users/richardkapolnai/go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/core.go:99 +0x12c
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0006941a0, {0xc000298180, 0x1, 0x1})
	/Users/richardkapolnai/go/pkg/mod/go.uber.org/zap@v1.27.0/zapcore/entry.go:253 +0x14c
go.uber.org/zap.(*Logger).Info(0xc00052d800, {0x103fc62f2, 0x18}, {0xc000298180, 0x1, 0x1})
	/Users/richardkapolnai/go/pkg/mod/go.uber.org/zap@v1.27.0/logger.go:247 +0x60
github.com/hslatman/caddy-crowdsec-bouncer/internal/bouncer.(*Bouncer).startMetricsProvider.func1()
	/Users/richardkapolnai/torolheto-privat/caddy-crowdsec-bouncer/internal/bouncer/metrics.go:62 +0x470
created by github.com/hslatman/caddy-crowdsec-bouncer/internal/bouncer.(*Bouncer).startMetricsProvider in goroutine 23
	/Users/richardkapolnai/torolheto-privat/caddy-crowdsec-bouncer/internal/bouncer/metrics.go:56 +0xe8
FAIL	github.com/hslatman/caddy-crowdsec-bouncer/internal/bouncer	2.043s
FAIL
```

--- a/internal/bouncer/bouncer_test.go
+++ b/internal/bouncer/bouncer_test.go
@@ -125,6 +125,8 @@ func TestStreamingBouncer(t *testing.T) {
 	b, err := newBouncer(t)
 	require.NoError(t, err)
 
+	t.Cleanup(func() { _ = b.Shutdown() })
+
 	// activate httpmock so that responses can be mocked
 	httpmock.Activate()
 	defer httpmock.DeactivateAndReset()

